onnxruntime/core/optimizer/transpose_optimizer/layout_tranformer_dev_notes.md

```diff
@@ -0,0 +1,53 @@
+# Layout Transformer Dev Notes
+
+## Overview
+ONNX standard assumes NCHW format for tensor layout. However, NCHW is not the best (perf efficient) format for all hardware types. Depending on the underlying hardware, to get the best perf we need to convert the model or in some cases part of the model which will be executed on the hardware from NCHW -> NHWC. Layout Transformer enables just this. It works with ONNX and ORT format models.
+
+*Note: Currently Layout Transformer works only for compiling EPs like NNAPI, QNN EP etc... More work is needed for it to be compatible with EPs like CPU and CUDA which use static kernel registration.*
```

Comment:
FWIW this is largely figured out in https://github.com/microsoft/onnxruntime/pull/11912/

GetCapability is always called twice if the EP asks for NHWC. The first call to GetCapability is purely for layout changes so we're only dealing with ONNX operators that the transpose operator knows about. The second call does things like convert a QDQ node group to a supported quantized op, or fusions such as Conv+activation. 

It's a little obtuse as to what should be done in the first call to GetCapability vs. the second, so ideally we can refine that a bit.

Reply:
I added more clarification around GetCapability based on behavior on master branch.